engine/src/org/pentaho/di/trans/steps/fileinput/BaseFileInputStep.java

```diff
@@ -427,9 +427,14 @@ protected void fillFileAdditionalFields( D data, FileObject file ) throws FileSy
     data.path = KettleVFS.getFilename( file.getParent() );
     data.hidden = file.isHidden();
     data.extension = file.getName().getExtension();
-    data.lastModificationDateTime = new Date( file.getContent().getLastModifiedTime() );
     data.uriName = file.getName().getURI();
     data.rootUriName = file.getName().getRootURI();
-    data.size = file.getContent().getSize();
+    if ( file.getType().hasContent() ) {
```

Comment:
I already created unit test for that, but I refactored unit tests for the new TextFileInput and CsvInput since Monday. I.e. unit test uses new code that not in git yet (see #1852). #1852 contains unit test against "Size" this issue. I will discuss order of commits with Michael.